### PR TITLE
Add PAM2020 BudgetsGroup Workflow

### PR DESCRIPTION
**Pull request by agustibr**
Related to `feat/improve-budgets-group-flow'

### CODE REVIEW

File: lib/decidim_staging/budgets_groups_workflow_pam2020.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    #
+    # Returns Object (Scope).
+    def user_authorization_scope
+      return unless user_authorization
+
+      # @user_authorization_scope ||= Decidim::Scope.find_by(
+      # id: user_authorization.metadata["scope_id"])
+
+      @user_authorization_scope ||= Decidim::Scope.find_by(
+        "name->>'ca' = '#{user_authorization.metadata['scope']}'"
```
Comment (agustibr) [inline lib/decidim_staging/budgets_groups_workflow_pam2020.rb:70]: The `Census` stores the `scope.name["ca"]` in the metadata.

We could migrate it to store the `scope.id`.

File: lib/decidim_staging/budgets_groups_workflow_pam2020.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      return unless user_authorization_scope
+
+      @user_scope_component ||= budgets.each do |component|
+        return component if component.scope == user_authorization_scope
+      end
+    end
+
+    # the user can vote on one of these out of its scope
+    # Returns Array (of Components)
+    def user_out_scope_components
```
Comment (agustibr) [inline lib/decidim_staging/budgets_groups_workflow_pam2020.rb:54]: Method not used, I'll remove it

File: config/initializers/decidim.rb [hunk 3]
```diff
@@ -130,3 +131,14 @@
 
 Rails.application.config.i18n.available_locales = Decidim.available_locales
 Rails.application.config.i18n.default_locale = Decidim.default_locale
+
+module Decidim::Budgets::Groups
```
Comment (leio10) [inline config/initializers/decidim.rb:135]: you can just add the new workflow, without monkey patching this:
`Decidim::Budgets::Groups.workflows[:pam20202] = DecidimStaging::BudgetsGroupsWorkflowPam2020`

File: lib/decidim_staging/budgets_groups_workflow_pam2020.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    def highlighted?(component)
+      component == user_scope_component unless voted?(user_scope_component)
+    end
+
+    # Can vote in the budget component in the user's scope
+    # and in an extra component out of its scope
+    def vote_allowed?(component, consider_progress = true)
+      return true if component == user_scope_component
+
+      extra_components = voted
```
Comment (leio10) [inline lib/decidim_staging/budgets_groups_workflow_pam2020.rb:18]: the name `extra_components` could be confusing, maybe something like `components_with_order` would be clear?

File: config/initializers/decidim.rb (hunk 3 above)
Comment (agustibr) [reply config/initializers/decidim.rb:135]: Oh, that's Great 🎉 !!

File: lib/decidim_staging/budgets_groups_workflow_pam2020.rb (hunk 4 above)
Comment (agustibr) [reply lib/decidim_staging/budgets_groups_workflow_pam2020.rb:18]: Thanks for the suggestion, I agree.

File: lib/decidim_staging/budgets_groups_workflow_pam2020.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module DecidimStaging
+  # Specific Workflow for Barcelona's 2020 PAM
+  class BudgetsGroupsWorkflowPam2020 < Decidim::Budgets::Groups::Workflows::Base
+    PAM2020AUTHORIZATIONHANDLER = 'dummy_census_authorization_handler'
+
+    # The budget component in the user's scope is highlighted.
+    def highlighted?(component)
+      component == user_scope_component unless voted?(user_scope_component)
```
Comment (leio10) [inline lib/decidim_staging/budgets_groups_workflow_pam2020.rb:10]: I'd change this with `... unless user_scope_component && voted?(user_scope_component)`

**Pull request merged**